server/src/main/scala/services/graphstages/ArrivalsGraphStage.scala

```diff
@@ -106,16 +106,9 @@ class ArrivalsGraphStage(name: String = "",
       }
     }
 
-    def purgeExpired(flightsById: Map[Int, Arrival]): Map[Int, Arrival] = {
-      val expired: Arrival => Boolean = Crunch.hasExpired(now(), expireAfterMillis, (a: Arrival) => a.PcpTime.getOrElse(0L))
-      flightsById.filterNot {
-        case (_, arrival) => expired(arrival)
-      }
-    }
-
     def mergeAllSourcesAndPush(baseArrivals: Map[Int, Arrival], forecastArrivals: Map[Int, Arrival], liveArrivals: Map[Int, Arrival]): Unit = {
       val newMerged = mergeArrivals(baseArrivals, forecastArrivals, liveArrivals)
-      val minusExpired = purgeExpired(newMerged)
+      val minusExpired = Crunch.purgeExpired(newMerged, (a: Arrival) => a.PcpTime.getOrElse(0L), now, expireAfterMillis)
```

Comment:
I presume we are not expiring flights with no PCP time - but then again I'm not sure why we were removing the flight in the first place.

Reply:
All flights have their PCP time populated as soon as they come in (lines 97 - 99), so the orElse shouldn't ever be triggered